Source/Scene/modelMaterialsCommon.js

```diff
@@ -0,0 +1,792 @@
+                                }
+                            }
+                        }
+
+                        return {
+                            attributes : Object.keys(primitive.attributes),
+                            jointCount : jointCount
+                        };
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Modifies gltf in place.
+     *
+     * @private
+     */
+    var modelMaterialsCommon = function(gltf) {
+        if (!defined(gltf)) {
+            return undefined;
+        }
+
+        var hasExtension = false;
+        var extensionsUsed = gltf.extensionsUsed;
+        if (defined(extensionsUsed)) {
+            var extensionsUsedCount = extensionsUsed.length;
+            for(var i=0;i<extensionsUsedCount;++i) {
+                if (extensionsUsed[i] === 'KHR_materials_common') {
+                    hasExtension = true;
+                    extensionsUsed.splice(i, 1);
+                    break;
+                }
+            }
+        }
+
+        if (hasExtension) {
+            if (!defined(gltf.programs)) {
+                gltf.programs = {};
+            }
+            if (!defined(gltf.shaders)) {
+                gltf.shaders = {};
+            }
+            if (!defined(gltf.techniques)) {
+                gltf.techniques = {};
+            }
+
+            var lightParameters = generateLightParameters(gltf);
+
+            var techniques = {};
+            var materials = gltf.materials;
+            var meshes = gltf.meshes;
+            for (var name in materials) {
+                if (materials.hasOwnProperty(name)) {
+                    var material = materials[name];
+                    var primitiveInfo = getPrimitiveInfo(name, gltf);
+                    if (defined(material.extensions) && defined(material.extensions.KHR_materials_common)) {
+                        var khrMaterialsCommon = material.extensions.KHR_materials_common;
+                        var key = getKey(khrMaterialsCommon);
+                        var technique = techniques[key];
+                        if (!defined(technique)) {
+                            technique = generateTechnique(gltf, khrMaterialsCommon, primitiveInfo.attributes,
+                                                          lightParameters, primitiveInfo.jointCount);
+                        }
+
+                        // Take advantage of the fact that we generate techniques that use the
+                        // same names as the extension values.
+                        material.values = khrMaterialsCommon.values;
+                        material.technique = technique;
+
+                        delete material.extensions.KHR_materials_common;
+                    }
+                }
+            }
+
+            if (defined(gltf.extensions)) {
+                delete gltf.extensions.KHR_materials_common;
+            }
+
+            //var json = JSON.stringify(gltf, null, 4);
```

Comment:
I'll remove once its not needed. While we are still messing with the extension spec and generating sample files I was gonna leave it there. I've used it pretty consistently all week.